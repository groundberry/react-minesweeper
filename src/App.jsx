import React, { Component } from 'react';
import Grid from './Grid';
import { startGame } from './utils';
import './App.css';

class App extends Component {
  constructor() {
    super();

    const level = 1;
    const grid = startGame(level);

    this.state = {
      grid,
    };
  }

  render() {
    const { grid } = this.state;

    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Minesweeper</h1>
        </header>
        <Grid grid={grid} />
      </div>
    );
  }
}

export default App;
