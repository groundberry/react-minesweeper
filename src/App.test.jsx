import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('renders the name of the app', () => {
    expect(wrapper.find('h1').text()).toEqual('Minesweeper');
  });

  it('renders a Face component', () => {
    expect(wrapper.find('LevelOptions')).toHaveLength(1);
  });

  it('renders a Face component', () => {
    expect(wrapper.find('Face')).toHaveLength(1);
  });

  it('renders a Grid component', () => {
    expect(wrapper.find('Grid')).toHaveLength(1);
  });
});
