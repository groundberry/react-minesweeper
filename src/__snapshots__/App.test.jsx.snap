// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the name of the app 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="App-title"
    >
      Minesweeper
    </h1>
  </header>
  <Grid
    grid={
      Array [
        Array [
          "0",
          "0",
          "0",
          "0",
          "0",
          "1",
          "*",
          "1",
          "0",
        ],
        Array [
          "0",
          "0",
          "0",
          "0",
          "0",
          "1",
          "1",
          "1",
          "0",
        ],
        Array [
          "0",
          "0",
          "1",
          "1",
          "1",
          "0",
          "0",
          "0",
          "0",
        ],
        Array [
          "0",
          "0",
          "1",
          "*",
          "1",
          "1",
          "1",
          "2",
          "1",
        ],
        Array [
          "1",
          "1",
          "1",
          "1",
          "1",
          "1",
          "*",
          "2",
          "*",
        ],
        Array [
          "*",
          "1",
          "0",
          "0",
          "0",
          "1",
          "1",
          "3",
          "2",
        ],
        Array [
          "1",
          "1",
          "0",
          "0",
          "0",
          "0",
          "0",
          "1",
          "*",
        ],
        Array [
          "1",
          "1",
          "1",
          "1",
          "2",
          "2",
          "1",
          "1",
          "1",
        ],
        Array [
          "1",
          "*",
          "1",
          "1",
          "*",
          "*",
          "1",
          "0",
          "0",
        ],
      ]
    }
  />
</div>
`;
